### PR TITLE
[WJ-268] Merge ftml into monorepo

### PR DESCRIPTION
**Pull request by emmiegit**
This fulfills [FTML-27](https://scuttle.atlassian.net/browse/FTML-27), required by [WJ-268](https://scuttle.atlassian.net/browse/WJ-268), which aims to rewrite the Wikidot parser with the goals of maintainability and ease of future extension.

In order to preserve history and context around changes, as well as provide a blame, all the commits from the [original repo](https://github.com/Nu-SCPTheme/ftml) have been grafted into `develop` here. The commit history was rewritten so that they only affect the top-level directory `ftml`, which will contain files for this service. As such, this should be a merge or rebase commit, _not_ a squash commit. Also please avoid rebasing the PR, as this will interfere with imported version tags.

As described on its [README](https://github.com/Nu-SCPTheme/ftml/blob/master/README.md), `ftml` is a Rust library to parse and render Wikidot text instead of hacking changes into `Text_Wiki`. It also has a sub-crate, `ftml-http`, which provides a simple REST server to wrap the main calls the library provides. The crate does not use any unsafe code and has several tests on its abstract syntax tree (AST) generation to prevent regressions.

Its progress has been tracked on the [FTML](https://scuttle.atlassian.net/browse/FTML) board for some time.

### CODE REVIEW

File: ftml/.github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+on:
```
Comment (rossjrw) [inline ftml/.github/workflows/ci.yaml]: I _think_ any Actions config needs to be in the top-level .github directory

**Comment by emmiegit** [comment]
https://scuttle.atlassian.net/browse/FTML-27

https://scuttle.atlassian.net/browse/WJ-268

**Review by rossjrw — APPROVED** [review]
It's good to see this chunk of the project joining the rest of its family.

File: ftml/.github/workflows/ci.yaml (hunk 1 above)
Comment (emmiegit) [reply ftml/.github/workflows/ci.yaml]: Yeah, this just copies all the commits from the repo, there are some things to do afterwards with adjusting CI and getting it integrated into Docker.

**Comment by emmiegit** [comment]
Fixed the CI, whenever the service is modified the builds will fire.

**Comment by emmiegit** [comment]
Documentation and such should now be updated.

File: ftml/docs/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+../README.md
```
Comment (rossjrw) [inline ftml/docs/README.md:1]: Curious - what's this link here for?

File: .github/workflows/test-js.yaml [hunk 3]
```diff
@@ -1,4 +1,5 @@
-name: '[Javascript] tests'
+name: '[js] Tests'
```
Comment (rossjrw) [inline .github/workflows/test-js.yaml]: Given that the FTML badge title is `[<project>] language`, I'd prefer something like `]frontend] JS` or `[web] JS` for consistency. The URL for the badge on the top README will also need to be updated

**Review by rossjrw — APPROVED** [review]

File: ftml/docs/Philosophy.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+Wikitext is similar to Markdown and dissimilar to C in that the grammar is loose.
+Any invalid token combinations are rendered as-is, rather than producing a fatal parsing
+error which halts the process. This latter, C-like (or any programming language, really)
+philosophy was how the original version of ftml operated. However this presents obvious
+incomaptibilities with Wikidot, and the grammar had to be increasingly complicated to handle
+edge-case conditions.
+
+This rewrite of ftml performs preprocessing substitions and tokenization like the first version,
+but the parser is hand-written to allow for loose fallback rules.
```
Comment (rossjrw) [inline ftml/docs/Philosophy.md]: I'm not sure the distinction between the current version of ftml and the original version, while historically important in terms of code history and developer experience, important reading for what I expect to be the average reader of this document. I feel like the philosophy of fallbacks could be described with the assumption that the current version is the only canonical version

File: ftml/docs/README.md (hunk 2 above)
Comment (emmiegit) [reply ftml/docs/README.md:1]: I guess since it's documentation. It's not really necessary, I just thought I'd add it.

**Comment by emmiegit** [comment]
thanks @rossjrw

**Pull request merged**